test/abstract_syntax_tree.coffee

```diff
@@ -838,6 +838,17 @@ test "AST as expected for Index node", ->
     optional: no
     shorthand: no
 
+  testExpression 'a[if b then c]',
+    type: 'MemberExpression'
+    object: ID 'a'
+    property:
+      type: 'ConditionalExpression'
```

Comment:
This test would have previously been breaking since the if statement AST was being generated at `LEVEL_TOP` (which was "bleeding through" from the outer context) so was producing a block with an `IfStatement` instead of a `ConditionalExpression`